### PR TITLE
THORN-2418: SSL support tests

### PR DESCRIPTION
**Pull request by penehyba**

### CODE REVIEW

File: microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/RestClientResource.java [hunk 1]
```diff
@@ -70,4 +77,42 @@ public String close() throws Exception {
...
+        try {
+            KeyStore trustStore = getKeyStore(truststoreInputStream);
+            KeyStore keyStore = getKeyStore(keystoreInputStream);
+
+
+            return RestClientBuilder.newBuilder()
+                    .baseUrl(new URL(HTTP_LOCALHOST_8080))
+                    .trustStore(trustStore)
+                    .keyStore(keyStore, PASSWORD)
+                    .hostnameVerifier(new AcceptAllVerifier())
```
Comment (Ladicek) [inline microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/RestClientResource.java]: This shouldn't be needed, the server certificate is issued for `localhost`.

File: microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/sslsupport/SslClient.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.wildfly.swarm.ts.microprofile.rest.client.v13.sslsupport;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@RegisterRestClient(configKey = "sslClient")
+public interface SslClient {
```
Comment (Ladicek) [inline microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/sslsupport/SslClient.java]: The `CdiSslClient` and `SslClient` interfaces are identical, there's no need to have 2 of them. Remove the `CdiSslClient` thing, it's superfluous.

File: microprofile/microprofile-rest-client-1.3/src/test/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/SslSupportTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @Test
+    @RunAsClient
+    public void testCdiClient() throws IOException {
+        String response = Request.Get("http://localhost:8080/rest/client/cdi-ssl-client").execute().returnContent().asString();
+        assertThat(response).isEqualTo("Hello from SSL cdi client.");
+    }
+
+    public static KeyStore getKeyStore(File keystoreFile) throws Exception {
```
Comment (Ladicek) [inline microprofile/microprofile-rest-client-1.3/src/test/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/SslSupportTest.java]: Is this method used somewhere? It doesn't seem to be used in this class.

File: microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/SimpleResource.java [hunk 4]
```diff
@@ -12,4 +12,16 @@
...
+
+    @GET
+    @Path("/ssl-client")
+    public Response simple() {
+        return Response.ok().entity("Hello from SSL client.").build();
+    }
+
+    @GET
+    @Path("/cdi-ssl-client")
+    public Response simpleCDI() {
```
Comment (Ladicek) [inline microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/SimpleResource.java]: I'd remove these 2 methods (`simple` and `simpleCDI`) and let the SSL client call `/rest/simple`. Then, you would prepend something like `"SSL client got: "` to the response the SSL client received, and send that back as your response.

File: microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/RestClientResource.java [hunk 5]
```diff
@@ -70,4 +77,42 @@ public String close() throws Exception {
...
+                    .build(SslClient.class).simple();
```
Comment (Ladicek) [inline microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/RestClientResource.java]: The call to `.simple()` should be on an extra line, it's the most important thing that happens here.

File: microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/RestClientResource.java (hunk 1 above)
Comment (penehyba) [reply microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/RestClientResource.java]: modified to org.apache.http.conn.ssl.DefaultHostnameVerifier

File: microprofile/microprofile-rest-client-1.3/src/test/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/SslSupportTest.java (hunk 3 above)
Comment (penehyba) [reply microprofile/microprofile-rest-client-1.3/src/test/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/SslSupportTest.java]: removed

**Comment by Ladicek** [comment]
Thanks!

**Pull request merged**